contracts/interfaces/IP2PPairStaking.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.8.10;
+
+import "../dependencies/openzeppelin/contracts/IERC20.sol";
+
+interface IP2PPairStaking {
+    enum StakingType {
+        BAYCStaking,
+        MAYCStaking,
+        BAYCPairStaking,
+        MAYCPairStaking
+    }
+
+    struct ListingOrder {
+        StakingType stakingType;
+        address offerer;
+        address token;
+        uint32 tokenId;
+        uint32 share;
+        uint256 startTime;
+        uint256 endTime;
+        uint8 v;
+        bytes32 r;
+        bytes32 s;
+    }
+
+    struct MatchedOrder {
+        StakingType stakingType;
+        address apeToken;
+        address apeOfferer;
+        uint32 apeTokenId;
```

Comment:
should we use uint16 here? 2^16 is 65,536 which is larger than all token ids in BAYC, MAYC and BAKC

Reply:
apeOfferer, apeTokenId, apeShare will share a slot no matter apeTokenId is uint16 or uint32.